src/sagemaker/serve/builder/model_builder.py

```diff
@@ -1302,6 +1302,10 @@ def _model_builder_optimize_wrapper(
         job_name = job_name or f"modelbuilderjob-{uuid.uuid4().hex}"
         if self._is_jumpstart_model_id():
             self.build(mode=self.mode, sagemaker_session=self.sagemaker_session)
+            if self.pysdk_model:
```

Comment:
Can we add a test for this in `test_model_builder` ?

Reply:
[tests/integ/sagemaker/serve/test_serve_js_deep_unit_tests.py](https://github.com/aws/sagemaker-python-sdk/pull/4937/files#diff-557ffe7eb4dcf1f09701ae8c60fbdb57e8d115913c2e5d4f6c5ad9de641a68aa) under `test_js_model_with_optimize_quantization_on_pre_optimized_model_requests_are_expected`